internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/DataRecoveryTest.java

```diff
@@ -234,27 +268,26 @@ public void testBatchOnlyDataRecovery() throws Exception {
           parentControllerClient,
           60,
           TimeUnit.SECONDS);
-      // Prepare dc-1 for data recovery
-      Assert.assertFalse(
-          parentControllerClient.prepareDataRecovery("dc-0", "dc-1", storeName, VERSION_ID_UNSET, Optional.empty())
+
+      // Verify that if same data center are given as both src and dest to data recovery api, client response should
+      // have errors.
+      String sameFabricName = "dc-0";
+      Assert.assertTrue(
+          parentControllerClient
+              .prepareDataRecovery(sameFabricName, sameFabricName, storeName, VERSION_ID_UNSET, Optional.empty())
               .isError());
-      // Initiate data recovery, a new version will be created in dest fabric
-      Assert.assertFalse(
+      Assert.assertTrue(
           parentControllerClient
-              .dataRecovery("dc-0", "dc-1", storeName, VERSION_ID_UNSET, false, true, Optional.empty())
+              .dataRecovery(sameFabricName, sameFabricName, storeName, VERSION_ID_UNSET, false, true, Optional.empty())
               .isError());
-      TestUtils.waitForNonDeterministicAssertion(60, TimeUnit.SECONDS, true, () -> {
-        Assert.assertEquals(dc1Client.getStore(storeName).getStore().getCurrentVersion(), 2);
-      });
-      try (AvroGenericStoreClient<String, Object> client = ClientFactory.getAndStartGenericAvroClient(
-          ClientConfig.defaultGenericClientConfig(storeName)
-              .setVeniceURL(childDatacenters.get(1).getClusters().get(clusterName).getRandomRouterURL()))) {
-        for (int i = 0; i < 10; i++) {
-          Object v = client.get(String.valueOf(i)).get();
-          Assert.assertNotNull(v, "Batch data should be consumed already in data center dc-1");
-          Assert.assertEquals(v.toString(), String.valueOf(i));
-        }
-      }
+
+      // Perform multiple rounds of dc-0 --> dc-1 data recovery.
+      performDataRecoveryTest(storeName, parentControllerClient, dc1Client, "dc-0", "dc-1", 2, 2);
+
+      // Perform multiple rounds of dc-1 --> dc-0 data recovery. Now, since src colo's (dc-1) current version is 3, we
+      // should expect
+      // dc-0 starts with dc-1's current version = 3;
+      performDataRecoveryTest(storeName, parentControllerClient, dc0Client, "dc-1", "dc-0", 2, 3);
```

Comment:
It took me some time to understand the test. Please check if my understanding is correct:

Before data recovery, current version in dc-0 and dc-0 is 1. With two rounds of dc-0 -> dc-1 data recovery, current version in dc-1 changes to 2 and then 3. Then with two rounds of dc-1 -> dc-0 data recovery, current version in dc-0 becomes 3 and then 4.

Reply:
Yes, your understanding is exactly right. If you find it not easy to understand, I should add more comments to make it clearer. In fact your writeup is actually a good one I think.